components/tag-list/tag-list.js

```diff
@@ -350,9 +352,11 @@ class TagList extends LocalizeCoreElement(InteractiveMixin(ArrowKeysMixin(LitEle
 		const itemIndex = children.indexOf(rootTarget);
 
 		if (children.length <= 1) return;
-		const focusableElem = children[itemIndex - 1] || children[itemIndex + 1];
```

Comment:
With the previous implementation, it looks like it was assuming that the consumer would not have removed the item yet, and then trying to either focus the previous item if it exists, otherwise the next item. I gather that was not working as intended. Was that assumption not valid, or was something else going wrong?

With the new implementation, it seems like the `children[itemIndex] !== e.target` is meant to handle both the case where the item has been removed by the time the callback runs, and the case where it hasn't been removed yet. Am I understanding that correctly?

Reply:
Yeah that's exactly right. In `tag-list` the item is not yet removed by the time it gets to this method, but in `filter-tags` it is. However, I realize now that this is simply because `children` is outdated. I think it'd be simpler to move everything inside the `setTimeout` and then do:
```
const focusableElem = children[itemIndex - 1] || children[0];
```

Reply:
Yeah that's exactly right. In `tag-list` the item is not removed until after the timeout, but in `filter-tags` it's removed before the method is called.

Reply:
lol, I keep writing comments and deleting them because I keep finding more weird stuff. It looks like this might be down to a lit DOM updating issue between `tag-list` and `filter-tags`. It's basically removing a different child from the DOM than what was deleted, so the references change, but of course the indexes don't.